src/framework/mpas_string_utils.F

```diff
@@ -35,14 +35,16 @@ subroutine mpas_split_string(string, delimiter, subStrings)
     implicit none
 
     character(len=*), intent(in) :: string
+    character(len=len(trim(string))) :: trimString
```

Comment:
I'd propose moving the declaration of this local variable down to around line 42 with other local variables. Generally (and we should add this to a coding guidelines document!), I think we'd like to have dummy argument declarations first, then some separation, then local variable declarations. In newer code (e.g., in [mpas_halo.F](https://github.com/MPAS-Dev/MPAS-Model/blob/v8.0.0/src/framework/mpas_halo.F#L118-L124)) we've gone so far as to add comments `! Arguments` and `! Local variables` to call out which declarations are which.

Reply:
Also, rather than `len(trim(string))` you can use `len_trim(string)`. Section 13 of the Fortran 2003 standard, which covers intrinsic procedures, is good reading if you're ever waiting on something to compile!